DependencyInjection/Repository/Factory/CompositeFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2016 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\ResourceBundle\DependencyInjection\Repository\Factory;
+
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Definition;
+use Puli\Repository\CompositeRepository;
+use Symfony\Cmf\Bundle\ResourceBundle\DependencyInjection\CmfResourceExtension;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+class CompositeFactory implements RepositoryFactoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $options)
+    {
+        $definition = new Definition(CompositeRepository::class);
+
+        foreach ($options['mounts'] as $mountConfig) {
+            $definition->addMethodCall(
+                'mount',
+                [
+                    $mountConfig['mountpoint'],
+                    new Reference(
+                        CmfResourceExtension::getRepositoryServiceId($mountConfig['repository'])
+                    ),
+                ]
+            );
+        }
+
+        return $definition;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'composite';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(OptionsResolver $options)
+    {
+        $options->setDefault('mounts', []);
+        $options->setAllowedTypes('mounts', ['array']);
```

Comment:
This can be just `setAllowedTypes('mounts', 'array')`, but it's not that important.